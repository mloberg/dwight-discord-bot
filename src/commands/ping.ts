import { SlashCommandBuilder } from '@discordjs/builders';
import { CommandInteraction } from 'discord.js';

export default {
    config: new SlashCommandBuilder().setName('ping').setDescription('Pong'),
    async handle(command: CommandInteraction): Promise<void> {
        await command.reply('🏓');
    },
};
