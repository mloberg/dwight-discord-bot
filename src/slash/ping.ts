import { SlashCommand } from '../types';

const ping: SlashCommand = {
    name: 'ping',
    description: 'pong',
    async run(command) {
        await command.reply('🏓');
    },
};

export default ping;
