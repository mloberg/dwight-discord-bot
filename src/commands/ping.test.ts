import { CommandInteraction } from 'discord.js';

import ping from './ping';

jest.mock('discord.js');

describe('/ping', () => {
    it('is a slash command', () => {
        expect(ping).toMatchSnapshot();
    });

    it('replies', async () => {
        const command = new CommandInteraction({} as never, {} as never);
        await ping.handle(command);
        expect(command.reply).toHaveBeenCalledWith('🏓');
    });
});
