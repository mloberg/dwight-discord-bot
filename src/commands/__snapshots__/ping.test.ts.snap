// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/ping is a slash command 1`] = `
Object {
  "config": Object {
    "default_permission": undefined,
    "description": "Pong",
    "name": "ping",
    "options": Array [],
  },
  "handle": [Function],
}
`;
